packages/ui/src/components/PropertiesModal/PropertiesModal.tsx

```diff
@@ -0,0 +1,76 @@
+import { Modal } from '@patternfly/react-core';
+import { Caption, Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { FunctionComponent } from 'react';
+import { camelComponentToTable, kameletToTable } from '../../camel-utils/camel-to-table.adapter';
+import { CatalogKind, ICamelComponentDefinition, ICamelProcessorDefinition, IKameletDefinition } from '../../models';
+import { ITile } from '../Catalog';
+import { EmptyTableState } from './EmptyTableState';
+import { IPropertiesTable, PropertiesHeaders } from './PropertiesTable.models';
+
+interface IPropertiesModalProps {
+  tile: ITile;
+  onClose: () => void;
+  isModalOpen: boolean;
+}
+
+export const PropertiesModal: FunctionComponent<IPropertiesModalProps> = (props) => {
+  let table: IPropertiesTable;
+  switch (props.tile.type) {
+    case CatalogKind.Component:
+      table = camelComponentToTable(props.tile.rawObject as ICamelComponentDefinition);
+      break;
+    case CatalogKind.Processor:
+      table = camelComponentToTable(props.tile.rawObject as ICamelProcessorDefinition);
+      break;
+    case CatalogKind.Kamelet:
+      table = kameletToTable(props.tile.rawObject as IKameletDefinition);
+      break;
+    default:
+      throw Error('Unknown CatalogKind during rendering modal: ' + props.tile?.type);
+  }
+
+  return (
+    <Modal title={props.tile.title} isOpen={props.isModalOpen} onClose={props.onClose} ouiaId="BasicModal">
+      <p>{props.tile.description}</p>
+      <Table aria-label="Simple table" variant="compact">
+        <Caption>{'Available properties (' + table.rows.length + ')'}</Caption>
+        <Thead>
+          <Tr>
+            {table.headers.includes(PropertiesHeaders.Name) && <Th>{PropertiesHeaders.Name}</Th>}
+            {table.headers.includes(PropertiesHeaders.Type) && <Th>{PropertiesHeaders.Type}</Th>}
+            {table.headers.includes(PropertiesHeaders.Kind) && <Th>{PropertiesHeaders.Kind}</Th>}
+            {table.headers.includes(PropertiesHeaders.DefaultValue) && <Th>{PropertiesHeaders.DefaultValue}</Th>}
+            {table.headers.includes(PropertiesHeaders.Required) && <Th>{PropertiesHeaders.Required}</Th>}
+            {table.headers.includes(PropertiesHeaders.Description) && <Th>{PropertiesHeaders.Description}</Th>}
+          </Tr>
+        </Thead>
+        <Tbody>
+          {table.rows.length != 0 &&
+            table.rows.map((row, index) => (
+              <Tr key={index}>
+                {table.headers.includes(PropertiesHeaders.Name) && (
```

Comment:
Probably we could iterate over the headers and extract the desired field, using something like: 

```typescript
row[PropertiesHeaders.Type]
```

Reply:
yes, it can be used that way, but the PropertiesHeaders.Type values needs to be match with object properties, so in lower case e.g.
```
export const enum PropertiesHeaders {
  Name = 'name',
  Type = 'type',
  ...
```

Reply:
refactored and force push